fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java

```diff
@@ -154,29 +155,26 @@ private void updatePartitionInfo(String databaseName, String tableName) {
 
         try {
             List<org.apache.paimon.partition.Partition> partitions = paimonNativeCatalog.listPartitions(identifier);
+            String partitionLegacyName = paimonTable.options().get(CoreOptions.PARTITION_GENERATE_LEGCY_NAME.key());
```

Comment:
the option shall be parsed here into a bool, and pass the bool to the getPartition() call, so the getPartition() don't have to parse the string again and again for every column for every partition.